test/tools/unified-spec-runner/entities.ts

```diff
@@ -395,10 +403,20 @@ export class EntitiesMap<E = Entity> extends Map<string, E> {
         map.set(entity.database.id, db);
       } else if ('collection' in entity) {
         const db = map.getEntity('db', entity.collection.database);
-        const collection = db.collection(
-          entity.collection.collectionName,
-          patchCollectionOptions(entity.collection.collectionOptions)
-        );
+        const collection = await db
```

Comment:
question: is this logic added to handle the scenario where a user has two entities with the same collection name in a single test suite?  If so, would it make more sense to flip the logic - list collections while querying for the collection, and then create it if it doesn't exist?

Reply:
Until this PR we haven't used the unified runner to run a drop db / rename collection. They both require the collection to exist in advance of the operations stage of the test. I could instead adjust the test to perform an insert to create the collection, but this seems like a reasonable adjustment to the runner.

Reply:
Oh that makes sense.  I was mainly wondering if the logic would be easier to follow if we flipped it around to look for collections first.  But I don't feel super strongly about this, feel free to ignore this thought

```typescript
let collection = await db.listCollections({ collectionName: entitiy.collection.collectionName })
    .map(({ name }) => db.collection(name))
    .toArray()[0]
if (!collection) {
  collection = db.createCollection( .... )
}
```